src/login/logind-button.c

```diff
@@ -307,55 +306,47 @@ static int button_set_mask(Button *b) {
                 .codes_ptr = PTR_TO_UINT64(switches),
         };
 
-        if (ioctl(b->fd, EVIOCSMASK, &mask) < 0)
-                return log_warning_errno(errno, "Failed to set EV_SW event mask on /dev/input/%s: %m", b->name);
+        if (ioctl(fd, EVIOCSMASK, &mask) < 0)
+                return log_warning_errno(errno, "Failed to set EV_SW event mask on /dev/input/%s: %m", name);
 
         return 0;
 }
 
 int button_open(Button *b) {
-        char *p, name[256];
+        _cleanup_close_ int fd = -1;
+        char name[256];
         int r;
 
         assert(b);
 
         b->fd = safe_close(b->fd);
 
-        p = strjoina("/dev/input/", b->name);
+        const char *p = strjoina("/dev/input/", b->name);
```

Comment:
Personally, I do not like this... Of course, our coding style now accepts this. Hmm...

Reply:
OK, I moved the variable definition to the top.